app/controllers/cart_controller.rb

```diff
@@ -8,15 +8,16 @@ def add_item
     else
       flash[:success] = "#{item.name} was successfully added to your cart"
       redirect_to items_path
-    end 
+    end
   end
 
   def show
+    non_current_user?
```

Comment:
nice addition of this method in the show action

Reply:
I like this non_current_user method too. More intuitive than current_user.nil?